docs/dns-records.md

```diff
@@ -0,0 +1,83 @@
+# Setting custom DNS records
+
+## Goal
+
+This documentation has the goal of showing how a user can set custom DNS records client with `headscale`s magic dns.
+An example usecase is to serve apps on the same host via a reverse proxy like NGinX, in this case a prometheus monitoring stack. This allows to nicely access the service with "http://grafana.myvpn.example.com" instead of the hostname and portnum combination "http://hostname-in-magic-dns.myvpn.example.com:3000".
```

Comment:
- `usecase` -> `use case`
- `NGinX` -> `NGINX `
- `prometheus` -> `Prometheus`
- `This allows to nicely access the service with "http://grafana.myvpn.example.com" instead of the hostname and portnum combination "http://hostname-in-magic-dns.myvpn.example.com:3000".` -> `This improves the user experience by providing access via custom hostname "http://grafana.myvpn.example.com" instead of access via the default hostname and port "http://hostname-in-magic-dns.myvpn.example.com:3000"`